### ISSUE TITLE
NameError: name 'os' is not defined while reading ../v8/build/common.gypi

### ISSUE THREAD

**Issue by karolherbst** [desura/Desurium#387]
there can be an error whily building cef.

https://gist.github.com/bafa97b231c64a259fb0

found here: https://bugs.gentoo.org/show_bug.cgi?id=395837#c30

**Comment by Smilex** [comment]
https://code.google.com/p/gyp/wiki/InputFormatReference#Predefined_Variables
This is relevant to the issue I believe.
"But other values may be encountered and this list should not be considered exhaustive. The gypd (debug) generator module does not provide a predefined value for OS. When invoking GYP with the gypd module, if a value for OS is needed, it must be provided on the command line, such as gyp -f gypd -DOS=mac."

**Comment by Smilex** [comment]
My observation of the pipeline to run the .gyp files, is that gyp_cef is run, which generates a string of arguments, etc. and passes it to the gyp.main. gyp.main is found in the **init**.py of the gyp module, located at build_cef/chromium-prefix/src/chromium/src/tools/gyp/pylib/gyp
My idea was to simply add some code into gyp_cef, which checks for your platform and adds the -DOS accordingly, as specified in https://code.google.com/p/gyp/wiki/InputFormatReference#Predefined_Variables . I'll paste the patch here, however I have my doubts that this fixes it. I also can't recreate the bug, so I can't test it.

-- PATCH: gyp_cef.patch

```
144a145,152
>   # Add -DOS based on platform. See https://code.google.com/p/gyp/wiki/InputFormatReference#Predefined_Variables
>   if sys.platform == 'linux2':
>         args.append('-DOS=linux')
>   else if sys.platform == 'darwin':
>         args.append('-DOS=mac')
>   else if sys.platform == 'win32':
>         args.append('-DOS=win')
>
```

**Comment by karolherbst** [comment]
I don't think, that this has anything to do with the -DOS flag, because "import os" isn't called at all. which is strange, because it should.

**Comment by karolherbst** [comment]
@hasufell
we need more information about your system. I don't know a way how to reproduce this bug.

**Comment by karolherbst** [comment]
It could be, that python3 is called although we wrapped python a bit. (system python is set to 3 via eselect)

**Comment by karolherbst** [comment]
I found this python3 patch on the chromium buglist: http://code.google.com/p/chromium/issues/attachmentText?id=61357&aid=613570010000&name=py3-2.diff&token=KzZelmF9QZn_Ohu_EiXtZqO0qCk%3A1359992019453

Line 10 is interesting, I guess I was right: "import os" isn't called at all. This bug occurs as long as python2 is not installed (not confirmed yet).

**Comment by karolherbst** [comment]
without python2 installed I got another error, but not this. With python3 the code can't be invoked, because of some other stuff

**Comment by TheSiege** [comment]
quoting a talk with hasufell:

```
[22:25] <hasufell> for python problem we have to use "python-any-r1"
[22:25] <hasufell> which exports python-any-r1_pkg_setup
[22:25] <hasufell> and then PYTHON_COMPAT=( python{2_6,2_7} ) before inherit
```

**Comment by hasufell** [comment]
correct use of python-any-r1 seems to fix this, can't reproduce anymore

**Comment by karolherbst** [comment]
fixed since 10cd9534f66ac51aa442dfec907e04c6b8ef5e4f
